modules/web-console/web-console-notification/src/main/java/org/apache/ignite/console/notification/services/MailService.java

```diff
@@ -0,0 +1,235 @@
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.stereotype.Service;
+
+/**
+ * Mail sending service.
+ */
+@Service
+public class MailService {
+    /** From email. */
+    @Value("${spring.mail.username}")
+    private String from;
+
+    /** From alias. */
+    @Value("${spring.mail.from.alias}")
+    private String fromAlias;
+
+    /** Expression parser. */
+    private static final ExpressionParser parser = new SpelExpressionParser();
+
+    /** Template parser context. */
+    private static final TemplateParserContext templateParserCtx = new TemplateParserContext("${", "}");
+
+    /** Message source. */
+    private MessageSource msgSrc;
+
+    /** JavaMail sender */
+    private JavaMailSender mailSnd;
+
+    /** Mail config. */
+    private MessagesProperties cfg;
+
+    /**
+     * @param msgSrc Message source.
+     * @param mailSnd Mail sender.
+     * @param cfg Mail properties.
+     */
+    public MailService(MessageSource msgSrc, JavaMailSender mailSnd, MessagesProperties cfg) {
+        this.msgSrc = msgSrc;
+        this.mailSnd = mailSnd;
+        this.cfg = cfg;
+    }
+
+    /**
+     * Send email.
+     *
+     * @param notification Notification.
+     */
+    public void send(Notification notification) throws IOException, MessagingException {
+        NotificationWrapper ctxObj = new NotificationWrapper(notification);
+
+        EvaluationContext ctx = createContext(ctxObj);
+
+        NotificationDescriptor desc = notification.getDescriptor();
+
+        ctxObj.setSubject(processExpressions(getMessage(desc.subjectCode()), ctx));
+        ctxObj.setMessage(processExpressions(getMessage(desc.messageCode()), ctx));
+
+        String template = loadMessageTemplate(notification.getDescriptor());
+
+        MimeMessage msg = mailSnd.createMimeMessage();
+
+        MimeMessageHelper msgHelper = new MimeMessageHelper(msg);
+
+        msgHelper.setTo(notification.getRecipient().getEmail());
+        msgHelper.setFrom(from, fromAlias);
+        msgHelper.setSubject(ctxObj.getSubject());
+        msgHelper.setText(template == null ? ctxObj.getMessage() : processExpressions(template, ctx), true);
+
+        mailSnd.send(msg);
+    }
+
+    /**
+     * @param type Notification type.
+     * @return Message template or empty string.
+     */
+    private String loadMessageTemplate(NotificationDescriptor type) throws IOException {
+        String path = cfg.getTemplatePath(type);
+
+        if (path == null)
+            return null;
+
+        URL url;
+
+        try {
+            url = new URL(path);
+        }
+        catch (MalformedURLException e) {
+            url = U.resolveIgniteUrl(path);
+
+            if (url == null)
+                url = new ClassPathResource(path).getURL();
+        }
+
+        try (Scanner scanner = new Scanner(url.openStream(), StandardCharsets.UTF_8.toString())) {
```

Comment:
Lets rework to File API?
Smth. like: `String content = new String(Files.readAllBytes(Paths.get(url)));`

Reply:
Fixed